devito/finite_differences/tools.py

```diff
@@ -261,3 +261,22 @@ def generate_indices_staggered(func, dim, order, side=None, x0=None):
             ind = [start, start - diff]
 
     return start, tuple(ind)
+
+
+def make_get_shifted_oringin(shift, shape):
+    """
+    Returns a callable that calculates a shifted origin for each derivative
+    of a given operation derivatives scheme (shape) and a shift object which
+    can be a None, a single value or a tuple with the given shape.
+    """
+    if shift is None:
+        return lambda s, d, i, j: None
+    if type(shift) is float:
+        return lambda s, d, i, j: d + s * d.spacing
+    if type(shift) is tuple and np.shape(shift) == shape:
+        if len(shape) == 1:
+            return lambda s, d, i, j: d + s[j] * d.spacing
+        if len(shape) == 2:
+            return lambda s, d, i, j: d + s[i][j] * d.spacing
```

Comment:
what happens if it's neither 1 nor 2 ?

are we missing tests?

Reply:
It will raise `ValueError` down there, but I do think we could have an extra `else` here and raise a value error with the proper justification.

Reply:
ah, yes, I think that's fine